mapentity/models.py

```diff
@@ -69,7 +71,71 @@ class MapEntityRestPermissions(rest_permissions.DjangoModelPermissions):
     }
 
 
-class BaseMapEntityMixin(models.Model):
+class DuplicateMixin(object):
+    can_duplicate = True
+
+    if can_duplicate:
+        def get_duplicate_url(self):
+            return reverse(self._entity.url_name(ENTITY_DUPLICATE), args=[str(self.pk)])
```

Comment:
Ithink it would be better to always define function but return None if can_duplicate is False